solidity/dashboard/src/App.js

```diff
@@ -16,19 +16,21 @@ const App = () => (
       <Web3ContextProvider>
         <ModalContextProvider>
           <ContractsDataContextProvider>
-            <SideMenuProvider>
-              <Router>
-                <main>
-                  <Header />
-                  <aside>
+            <HeaderContextProvider>
+              <SideMenuProvider>
+                <Router>
+                  <main>
+                    <Header />
                     <SideMenu />
-                  </aside>
-                  <div className="content">
-                    <Routing />
-                  </div>
-                </main>
-              </Router>
-            </SideMenuProvider>
+                    <div className="content-wrapper" id="main-content-wrapper">
```

Comment:
In my opinion this
```
grid-template-areas:
    "header main" auto
    "footer main" max-content / 200px auto;
```
does not reflect the structure of the app in a right way.

I think we ca simplify and just `grid-template-columns: @side-menu-width 1fr`, and we get 2 columns layout.

What do you think about something like this?
```
<div class="wrapper"> // grid-template-columns: @side-menu-width 1fr
 <div class="side-menu">
  <header/>
  <nav/>
  <footer/>
 </div>
 <div class="content">
  <header/> <- as header component with page title and sublinks
  <main>
   page content here
  </main>
 </div>
</div>
```

Reply:
I think this achieves the visual aim but continues to leave the semantic hierarchy muddled. Can you explain how the structure of the app isn't reflected in my suggestion? It feels like your suggested revision tries to align the semantic hierarchy to the visual structure in a way that isn't necessary.

I find it helpful to think about two aspects of this when I'm trying to figure out the semantic hierarchy. The first is, “how would a typical page present this?” The second and most important one is “how should this flow for someone who cannot see the visual arrangement that we've chosen”. In that context, the `footer` is always the least important and last thing you want presented to someone browsing the structure (the most typical example being a user with a screen reader), because it's the one they're least likely to care about. The header and nav tend to be near the top of the ordering because they're the ones that tell the user the most information about what they're looking at and how the site hierarchy can be navigated.

This is where my proposed page structure comes from: it puts the site title and nav and page title and nav front and center, and finally puts the main content before the footer.

This also feeds my own driving principle for semantics, which is that a page that is well-structured *semantically* will also be easier to manage *programmatically*. In this case, I think a structure like this also allows the data flow to be hierarchical in more places, as opposed to having to use contexts to manage data flow non-hierarchically. Of course, this assumes hierarchical data flow is better—which I also happen to believe, but am open to debate on :)

Overall I still think my original suggestion has advantages over the current state, but I'm also not willing to block what we have here. It's solid work, and it looks and works well. Let's continue the conversation moving forward, but separately.

Reply:
For me the confusing part was:
```
grid-template-areas:
    "header main" auto
    "footer main" max-content / 200px auto;
```

since `header` it's not a `header` but `side bar`<- in visual structure. In my case, I thinking of the header as a container on the top, not on the left or right side. I think the grid templates should reflect the visual structure since its better to understand what's going on.

The more I wonder the more i understand your point. As I understand we want in React component something like this:

```
header that contains
 app-logo
 nav
routing- that renders the page content with sub heder and nav
footer
```
But we want to deal with element positioning in the css side.

It all depends on the approach that we use. I tried to align React components to what we see on the screen, but agree with your principle for semantics.

I'll open a new PR in a while to achive your semathic structure.